File: lib-database/src/main/java/com/ustadmobile/core/db/dao/ScrapeRunDao.java
package com.ustadmobile.core.db.dao;

import com.ustadmobile.lib.database.annotation.UmDao;
import com.ustadmobile.lib.database.annotation.UmQuery;
import com.ustadmobile.lib.db.entities.ScrapeRun;
import com.ustadmobile.lib.db.sync.dao.BaseDao;

@UmDao
public abstract class ScrapeRunDao implements BaseDao<ScrapeRun> {

    public static final String SCRAPE_TYPE_KHAN = "khan";

    public static final String SCRAPE_TYPE_VOA = "voa";

    public static final String SCRAPE_TYPE_EDRAAK = "edraak";

    @UmQuery("SELECT scrapeRunUid From ScrapeRun WHERE scrapeType = :scrapeType LIMIT 1")
    public abstract int findPendingRunIdByScraperType(String scrapeType);
}

File: lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/ScrapeRunDao.kt
package com.ustadmobile.core.db.dao

import com.ustadmobile.door.annotation.Dao
import androidx.room.Query
import com.ustadmobile.lib.db.entities.ScrapeRun

@Dao
abstract class ScrapeRunDao : BaseDao<ScrapeRun> {

    companion object {

        const val SCRAPE_TYPE_KHAN = "khan"

        const val SCRAPE_TYPE_VOA = "voa"

        const val SCRAPE_TYPE_EDRAAK = "edraak"

        const val SCRAPE_TYPE_GDL = "gdl"

        const val SCRAPE_TYPE_CK12 = "ck12"

        const val SCRAPE_TYPE_DDL = "ddl"
    }
}
